docs/tutorials/2. Real-time analysis.md

```diff
@@ -0,0 +1,225 @@
+
+```javascript
+window.onload = () => {
+    recordButton.onclick = function() {
+        var recording = this.classList.contains("recording");
+        if (!recording) {
+            this.setAttribute("disabled", true);
+
+            EssentiaWASM().then(function(essentiaModule) {
+                if (!isEssentiaInstance) {
+                    essentia = new Essentia(essentiaModule);
+                    isEssentiaInstance = true;
+                }
+                startMicRecordStream(enableButton); // `enableButton` is just a function that re-enables the Start/Stop button
+            });
+        } else {
+            stopMicRecordStream();
+        }
+    };
+}
+```
+
+For a more full-fledged example, take a look at our [HPCP Chroma RT demo](https://github.com/MTG/essentia.js/blob/dev/examples/hpcp-chroma-rt/index.html), which uses this technique.
+
+---
+
+## Using the Web Audio API with [AudioWorklets](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorklet).
+
+Another option for high-performance real-time audio on the web, more efficient, and non-deprecated, is AudioWorklets. Our [online demo for real-time RMS](https://mtg.github.io/essentia.js/examples/rms-rt/) level detection uses this approach. You can check out the code [here](https://github.com/MTG/essentia.js/tree/dev/examples/rms-rt), but the following snippets will go over the main differences from using `ScriptProcessorNode`:
+
+1. Getting a live audio stream from the user's microphone is done as above (using `navigator.mediaDevices.getUserMedia`), as is setting up the audio graph, connecting the nodes.
+2. When setting up the audio graph, we have to create an [`AudioWorkletNode`](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletNode) instead of a `ScriptProcessorNode`. This will allow us to run our custom audio code on its own rendering thread, separate from the UI. The following code returns one such node:
+
+```javascript
+const workletProcessorCode = ["https://cdn.jsdelivr.net/npm/essentia.js@0.1.0/dist/essentia-wasm.module.js", "https://cdn.jsdelivr.net/npm/essentia.js@0.1.0/dist/essentia.js-core.es.js", "essentia-worklet-processor.js"];
+
+export async function createEssentiaNode (audioCtx) {
+  try {
+    let concatenatedCode = await URLFromFiles(workletProcessorCode)
+    await audioCtx.audioWorklet.addModule(concatenatedCode); // add our custom code to the worklet scope
+  } catch(e) {
+    console.log(e);
+  }
+  return new AudioWorkletNode(audioCtx, 'essentia-worklet-processor');
+}
+```
+
+3. The async `URLFromFiles()` function ([by Paul Adenot](https://github.com/padenot/ringbuf.js/blob/master/example/utils.js)) is used to fetch and concatenate our custom code together with the `essentia.js` library. This is necessary because the `fetch` or `XHR` APIs are not available in the Worklets' scope, and ES6 module imports are only available in Chrome. This code allows our custom processing to work also on Firefox and Edge.
```

Comment:
Rephrase to -> "The async `URLFromFiles()` function used in ([ringbuf.js](https://github.com/padenot/ringbuf.js/blob/master/example/utils.js)) can be used ...."